src/draw/nema_gfx/lv_nema_gfx_path.c

```diff
@@ -0,0 +1,151 @@
+/**
+ * MIT License
+ *
+ * -----------------------------------------------------------------------------
+ * Copyright (c) 2008-24 Think Silicon Single Member PC
+ * -----------------------------------------------------------------------------
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next paragraph)
+ * shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/**
+ * @file lv_nema_gfx_path.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../core/lv_refr.h"
+
+#if LV_USE_NEMA_GFX
+#if LV_USE_NEMA_VG
+
+#include "lv_nema_gfx_path.h"
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+static int data_point = 0;
+static int seg_point  = 0;
+
+lv_nema_gfx_path_t * lv_nema_gfx_path_create(void)
+{
+    LV_PROFILER_BEGIN;
+    lv_nema_gfx_path_t * nema_gfx_path = lv_malloc_zeroed(sizeof(lv_nema_gfx_path_t));
+    LV_ASSERT_MALLOC(nema_gfx_path);
+    nema_gfx_path->seg = NULL;
+    nema_gfx_path->data = NULL;
+    nema_gfx_path->seg_size = 0;
+    nema_gfx_path->data_size = 0;
+    data_point = 0;
+    seg_point  = 0;
+    LV_PROFILER_END;
+    return nema_gfx_path;
+}
+
+void lv_nema_gfx_path_alloc(lv_nema_gfx_path_t * nema_gfx_path)
+{
+    LV_PROFILER_BEGIN;
+    nema_gfx_path->path = nema_vg_path_create();
+    nema_gfx_path->paint = nema_vg_paint_create();
+    nema_gfx_path->data = (float *) lv_malloc(nema_gfx_path->data_size * sizeof(float));
```

Comment:
The `lv_array` can be used here to eliminate the global variable `data_point ` and `seg_point `.